### PR TITLE
Add lowercase lamda

### LINKED ISSUE #3381: [BUG] [SPRING-CLOUD] Invalid generated ConfigurationProperties annotation
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

The generated Client Configurations class generates invalid properties annotation, when you use a title with uppercase characters. Camel case is not allowed for this annotation.

The yml
```
info:
  version: v1
  title: My Default API
```

Output is:
```
  @Bean
  @ConditionalOnProperty("myTestApi.security.macma.client-id")
  @ConfigurationProperties("myDefaultApi.security.x")
  public ClientCredentialsResourceDetails macmaResourceDetails() {
    ClientCredentialsResourceDetails details = new ClientCredentialsResourceDetails();
    details.setAccessTokenUri("test");
    return details;
  }
```

`@ConfigurationProperties("myDefaultApi.security.x")` is not allowed

see https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html
```
The prefix value for the annotation must be in kebab case (lowercase and separated by -, such as acme.my-project.person).

```

##### openapi-generator version

4.0.3

##### Suggest a fix

Camel case title should be seperated by a dot with just lower case characters or should be completely lower case.
```my.default.api.security.x```

or
```mydefaultapi.security.x```

### PR DESCRIPTION
**Pull request by SirWayne**
fixes #3381

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` . Default: `master`.
- [x] @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04)
--

### Description of the PR

Fix invalid spring properties prefix

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-cloud/clientConfiguration.mustache [hunk 1]
```diff
@@ -31,7 +31,7 @@ public class ClientConfiguration {
   private String {{{name}}}Password;
 
   @Bean
-  @ConditionalOnProperty(name = "{{{title}}}.security.{{{name}}}.username")
+  @ConditionalOnProperty(name = "{{#lowercase}}{{{title}}}{{/lowercase}}.security.{{{name}}}.username")
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-cloud/clientConfiguration.mustache]: shouldn't the lambda be invoked with `{{#lambda.lowercase}}`, as described in https://github.com/OpenAPITools/openapi-generator/blob/master/docs/templating.md#mustache-lambdas ?

**Comment by SirWayne** [comment]
@macjohnny Will this work?

```
 {{=<% %>=}}@Value("${{{#lambda.lowercase}}<%title%>{{/lambda.lowercase}}.security.<%name%>.username:}")<%={{ }}=%>
  private String {{{name}}}Username;
```

**Comment by macjohnny** [comment]
> Will this work?

I guess it wont, because the `{{=<% %>=}}` indicates erb style tags, see http://mustache.github.io/mustache.5.html

maybe this should be converted to normal mustache tags, and escape the `{` and `}`

**Comment by SirWayne** [comment]
Ok got it. Yes it would be better to escape the `{`, but there is no default way to escape braces.

It think a variable is a good solution:
```
 public char openbrace= `{`;
 public char closebrace= `{`;
```

Then i could use:
```
@Value("${{openBrace}}{#lambda.lowercase}}{{{title}}}{{/lambda.lowercase}}.security.{{{name}}}.username:{{closebrace}}")
```

Do you have a proposal,where i should add the `openbrace` variable?

**Comment by macjohnny** [comment]
how about
```mustache
 @Value("${{=<% %>=}}{<%={{ }}=%>{{#lambda.lowercase}}{{{title}}}{{/lambda.lowercase}}.security.<%name%>.key:{{=<% %>=}}}<%={{ }}=%>")
 private String {{{name}}}Key; 
```

**Comment by SirWayne** [comment]
This will work, but is not really readable. 
That's why I like the workaround with the variable. It's readable and easy to understand.

**Comment by macjohnny** [comment]
ok, a variable is a good solution. you can add it e.g. here
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java#L115

**Comment by wing328** [comment]
@SirWayne thanks for the PR, which has been included in the 4.1.0 release: https://twitter.com/oas_generator/status/1160000504455319553

**Pull request merged**